managers/oclc_catalog.py

```diff
@@ -0,0 +1,79 @@
+import os
+import requests
+from requests.exceptions import Timeout, ConnectionError
+
+from logger import createLog
+from managers.oclc_auth import OCLCAuthManager
+
+
+logger = createLog(__name__)
+
+
+class OCLCCatalogManager:
+    CATALOG_URL = 'http://www.worldcat.org/webservices/catalog/content/{}?wskey={}'
+    OCLC_SEARCH_URL = 'https://americas.discovery.api.oclc.org/worldcat/search/v2/'
+
+    def __init__(self):
+        self.oclcKey = os.environ['OCLC_API_KEY']
+        self.attempts = 0
+
+    def query_catalog(self, oclcNo):
+        catalog_response = None
+        self.attempts += 1
+        catalog_query = self.CATALOG_URL.format(oclcNo, self.oclcKey)
+        
+        if self.attempts > 3:
+            return catalog_response
+
+        try:
+            catalog_response = requests.get(catalog_query, timeout=3)
+        except (Timeout, ConnectionError):
+            logger.warn(f'Failed to query URL {catalog_query}')
+            return self.query_catalog(oclcNo)
+
+        if catalog_response.status_code != 200:
+            logger.warn(f'OCLC Catalog Request failed with status {catalog_response.status_code}')
+            return None
+
+        return catalog_response.text
+    
+    def query_other_editions(self, oclc_number: str): 
```

Comment:
I received a TypeError for this method when running the integration test.

 def test_query_brief_bibs(self, test_instance: OCLCCatalogManager):
>       assert test_instance.query_brief_bibs('ti:The Waves')['numberOfRecords'] > 1000
E       TypeError: 'NoneType' object is not subscriptable

Reply:
The test passes now with the new OCLC credentials